src/components/pages/RoleSignup/Applications/Mentee.js

```diff
@@ -40,35 +41,38 @@ const initialFormValues = {
   low_income: false,
   formerly_incarcerated: false,
   list_convictions: '',
-  subject: '',
+  subject: 'not collecting this from intake form',
   experience_level: '',
   job_help: false,
   industry_knowledge: false,
   pair_programming: false,
   other_info: '',
 };
 
-const Mentee = () => {
-  const [formValues, handleChange] = useForms(initialFormValues);
-  const [error, setError] = useState('');
-
+const Mentee = ({ dispatch, error, successPage }) => {
+  const [formValues, handleChange, _, setFormValues] =
+    useForms(initialFormValues);
   const history = useHistory();
 
-  const postNewAccount = async newAccount => {
-    try {
-      await axios.post(
-        `${process.env.REACT_APP_API_URI}/application/new/mentee`,
-        newAccount
-      );
-      history.push('/apply/success');
-    } catch (err) {
-      setError(err);
+  useEffect(() => {
+    axiosWithAuth()
+      .get(`/profile/current_user_profile`)
+      .then(res => {
+        setFormValues({ ...formValues, profile_id: res.data.profile_id });
+      })
+      .catch(err => {
+        console.error(err);
+      });
+
+    if (successPage) {
+      history.pushState(successPage);
+    } else if (error) {
+      console.error(error);
     }
-  };
+  }, [successPage, error]);
```

Comment:
Another linter, what can we do to get rid of this linter? Are we able to add the missing dependencies somehow?

Reply:
I can add the history dependency, however adding formValues and setFormValues causes an infinite loop because it will infinitely find and update the profile_id. I am not sure why the linter is complaining about this since it will break it if I do what it is asking.